### PR TITLE
fix:(rico-data)Implement missing mapper method toBean

### PR DESCRIPTION
**Pull request by madmas**
Implement missing mapper method in rico-data
also fix typos; one in an API method name, but as part of "incubation" package

### CODE REVIEW

File: incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java [hunk 1]
```diff
@@ -84,11 +84,29 @@ public BeanMapperImpl(final BeanManager beanManager) {
...
 
     @Override
     public <ID extends Serializable, B, E extends DataWithId<ID>> B toBean(final E entity, final Class<B> beanClass) {
-        throw new RuntimeException("Not yet implemented!");
+        Assert.requireNonNull(entity, "entity");
+        Assert.requireNonNull(beanClass, "beanClass");
+
+        final Class<E> entityClass = (Class<E>) entity.getClass();
+
+        B bean = beanManager.create(beanClass);
```
Comment (hendrikebbers) [inline incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java]: final

File: incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java [hunk 2]
```diff
@@ -84,11 +84,29 @@ public BeanMapperImpl(final BeanManager beanManager) {
...
+
+        Map<Class<?>, BeanConverter> converterMap = converters.computeIfAbsent(entityClass, c -> new HashMap<>());
```
Comment (hendrikebbers) [inline incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java]: final

File: incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java [hunk 3]
```diff
@@ -84,11 +84,29 @@ public BeanMapperImpl(final BeanManager beanManager) {
...
+        BeanConverter beanConverter = converterMap.get(beanClass);
```
Comment (hendrikebbers) [inline incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java]: final

**Review by hendrikebbers — COMMENTED** [review]
minor changes

File: incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java (hunk 2 above)
Comment (madmas) [reply incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java]: fixed

File: incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java (hunk 3 above)
Comment (madmas) [reply incubation/rico-data/src/main/java/dev/rico/internal/server/data/mapping/BeanMapperImpl.java]: fixed

**Comment by hendrikebbers** [comment]
LGTM

**Pull request merged**